### PR TITLE
Adts 28

### PR DESCRIPTION
**Pull request by selimcr**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -372,6 +379,12 @@ class App extends Component {
       () => this.setCurrentContrast());
   }
 
+  toggleEnableCustomLayouts = enableCustomLayouts => {
+    console.log("toggleEnableCustomLayouts in App", enableCustomLayouts);
```
Comment (jamescarpinter) [inline client/src/App.js]: We can probably remove this log item.

File: client/src/components/GraphViewerComponent/GraphViewerCytoscapeComponent/GraphViewerCytoscapeComponent.js [hunk 2]
```diff
@@ -474,7 +485,25 @@ export class GraphViewerCytoscapeComponent extends React.Component {
...
+  saveLayout() {
+    storageService.saveGraphLayoutPositions(this.layout, this.query, sessionService.getCurrentGraphLayoutPositions(this.layout, this.query));
+    const msg = `${this.layout} layout saved`;
+    print(msg);
+    this.setState({ notificationMessageIsVisible: true, notificationMessage: msg });
+  }
+
+  clearLayout() {
+    sessionService.clearGraphLayout(this.layout, this.query);
+    storageService.clearGraphLayout(this.layout, this.query);
```
Comment (jamescarpinter) [inline client/src/components/GraphViewerComponent/GraphViewerCytoscapeComponent/GraphViewerCytoscapeComponent.js]: Why are we storing the layout in two places?

Maybe we need to have a single service that wraps the interaction between these two services? I.e. a LayoutStorageService or similar?

File: client/src/components/ModelGraphViewerComponent/ModelGraphViewerComponent.js [hunk 3]
```diff
@@ -302,7 +302,7 @@ class ModelGraphViewerComponent extends React.Component {
       await this.cyRef.current.doLayout(this.progressCallback);
     } else {
       this.cyRef.current.addRelationships(this.componentRelationships, "component");
-      await this.cyRef.current.doLayout(this.progressCallback);
+      await this.cyRef.current(this.progressCallback);
```
Comment (jamescarpinter) [inline client/src/components/ModelGraphViewerComponent/ModelGraphViewerComponent.js]: Is this right? We can call current as a function?

File: client/src/components/ModelGraphViewerComponent/ModelGraphViewerCytoscapeComponent/ModelGraphViewerCytoscapeComponent.js [hunk 4]
```diff
@@ -153,7 +158,30 @@ export class ModelGraphViewerCytoscapeComponent extends React.Component {
...
+  saveLayout() {
+    storageService.saveModelsLayoutPositions(sessionService.getModelsLayoutPositions());
+    const msg = `${this.layout} layout saved`;
+    print(msg);
+    this.setState({ notificationMessageIsVisible: true, notificationMessage: msg });
+  }
+
+  clearLayout() {
+    sessionService.clearModelsLayout(this.layout);
+    storageService.clearModelsLayout(this.layout);
```
Comment (jamescarpinter) [inline client/src/components/ModelGraphViewerComponent/ModelGraphViewerCytoscapeComponent/ModelGraphViewerCytoscapeComponent.js]: As per earlier comment, we're saving in two places?

**Comment by jamescarpinter** [comment]
@jamiewilbraham I have done a quick look through this PR and no major issues from my perspective. A few suggestions though. Can take another look once you've done a smoke test and fixes have been made!

File: client/src/components/ModelGraphViewerComponent/ModelGraphViewerComponent.js (hunk 3 above)
Comment (JoseZarmada) [reply client/src/components/ModelGraphViewerComponent/ModelGraphViewerComponent.js]: You are completly right I debugged the functionality and manage to display this error when the component was called as a function 
image

File: client/src/App.js (hunk 1 above)
Comment (JoseZarmada) [reply client/src/App.js]: Removed Comment

File: client/src/components/ModelGraphViewerComponent/ModelGraphViewerCytoscapeComponent/ModelGraphViewerCytoscapeComponent.js (hunk 4 above)
Comment (JoseZarmada) [reply client/src/components/ModelGraphViewerComponent/ModelGraphViewerCytoscapeComponent/ModelGraphViewerCytoscapeComponent.js]: Created LayoutService class to handle the calls of each respective component in a unified class

File: client/src/components/GraphViewerComponent/GraphViewerCytoscapeComponent/GraphViewerCytoscapeComponent.js (hunk 2 above)
Comment (JoseZarmada) [reply client/src/components/GraphViewerComponent/GraphViewerCytoscapeComponent/GraphViewerCytoscapeComponent.js]: Created LayoutService class to handle the calls of each respective component in a unified class

**Pull request merged**